From d4ef7419e1a6532756c92c29339fa2d468b6adb8 Mon Sep 17 00:00:00 2001
From: Sugar Zhang <sugar.zhang@rock-chips.com>
Date: Fri, 3 Mar 2023 10:35:36 +0800
Subject: [PATCH 79/94] ASoC: rockchip: hdmi: Fix potential NULL point
 reference

Signed-off-by: Sugar Zhang <sugar.zhang@rock-chips.com>
Change-Id: I9d0c2adbdffc268d88bac31c1db31507fd82661d
Signed-off-by: Detlev Casanova <detlev.casanova@collabora.com>

Gbp-Pq: Name 0160-ASoC-rockchip-hdmi-Fix-potential-NULL-point-referenc.patch
---
 sound/soc/rockchip/rockchip_hdmi.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sound/soc/rockchip/rockchip_hdmi.c b/sound/soc/rockchip/rockchip_hdmi.c
index 449af7c81..395f8dbdf 100644
--- a/sound/soc/rockchip/rockchip_hdmi.c
+++ b/sound/soc/rockchip/rockchip_hdmi.c
@@ -227,6 +227,9 @@ static int rk_hdmi_probe(struct platform_device *pdev)
 
 	codecs = devm_kcalloc(&pdev->dev, idx,
 			      sizeof(*codecs), GFP_KERNEL);
+	if (!codecs)
+		return -ENOMEM;
+
 	rk_data->dai.codecs = codecs;
 	rk_data->dai.num_codecs = idx;
 	idx = 0;
-- 
2.46.0

